Bottom: ddfb232cd8662ca03dc1fe9224976120a33be45f
Top:    711d46ca9c18dfcafbe49829cde838c60cc9598c
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-22 21:52:18 +0000

hid/gtk: (GL) Disable scissor test for pinout preview widget.

The partial drawing is interacting badly with our double-buffering,
as we are not painting the entire buffer we are swapping onto the
screen.

I have left a comment and some debug code, as this may need revisiting
to address what is really going on here.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index ace9851..e35f11e 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1095,10 +1095,18 @@ ghid_pinout_preview_expose (GtkWidget *widget,
 
   glViewport (0, 0, allocation.width, allocation.height);
 
+#if 0
   glEnable (GL_SCISSOR_TEST);
   glScissor (ev->area.x,
              allocation.height - ev->area.height - ev->area.y,
              ev->area.width, ev->area.height);
+  printf ("EVT: %i, %i, w=%i, h=%i, Scissor setup: glScissor (%f, %f, %f, %f);\n",
+          ev->area.x, ev->area.y, ev->area.width, ev->area.height,
+             (double)ev->area.x,
+             (double)(allocation.height - ev->area.height - ev->area.y),
+             (double)ev->area.width,
+             (double)ev->area.height);
+#endif
 
   glMatrixMode (GL_PROJECTION);
   glLoadIdentity ();
